### ISSUE TITLE
Can PushAsync on Open Notification. Xamarin Forms Android. Error Getting Google Registration ID

### ISSUE THREAD

**Issue by mtoribio** [OneSignal/OneSignal-Xamarin-SDK#35]
Hi,
I'm trying to push a page after notification is open.
 #### Env ####
Xamarin.Forms 2.3.4
Xamarin.GooglePlayServices.Gcm 42.1001.0
Com.OneSignal 3.0.1
#### Targets ####
Android API 25 & 24
IOS 9.3 (Works)
#### Code #####
`App.xaml.cs`
```
public App()
{
    InitializeComponent();
    MainPage = new NavigationPage(new MenuPage()); 
    OneSignal.Current
        .StartInit("ccb831eb-0b48-4f96-b09b-87f2d70a8424")
        .HandleNotificationOpened(HandleNotificationOpened)
        .EndInit();
}
```
I am trying:

1. 

```
async void HandleNotificationOpened(OSNotificationOpenedResult result)
{
     await MainPage.Navigation.PushAsync(new ShowPage());
}
```

2. 

```
async void HandleNotificationOpened(OSNotificationOpenedResult result)
{
     await Task.Delay(1000);
     await MainPage.Navigation.PushAsync(new ShowPage());
}
```

#### Error ####
```
[OneSignal] Error Getting Google Registration ID
[OneSignal] java.io.IOException: INVALID_PARAMETERS
[OneSignal] 	at com.google.android.gms.gcm.GoogleCloudMessaging.zza(Unknown Source)
[OneSignal] 	at com.google.android.gms.gcm.GoogleCloudMessaging.register(Unknown Source)
[OneSignal] 	at com.onesignal.PushRegistratorGPS$2.run(PushRegistratorGPS.java:151)
[OneSignal] 	at java.lang.Thread.run(Thread.java:761)
```

I don't know if this is the issue because de app continue working, I mean the constructor of ShowPage work and the app doesn't crash.

On IOS works.

Thank you.

**Comment by forumd** [comment]
I tried to change the versions from _googleplayservices.Gcm_ unsuccessfully.
versions:
- 27.0.0.0
- 42.1001.0
- 32.961.0

**Comment by jkasten2** [comment]
@forumd If everything is working correctly for you then this error is from our SDK trying to re-register from a GCM token but not reusing the Google Project number.

See issue https://github.com/OneSignal/OneSignal-Android-SDK/issues/165 fro more details. It has been fixed in the native SDK and will be included in the next Xamarin release.

**Comment by xamarindevelopervietnam** [comment]
be quickly, please!

**Comment by rabih-harb** [comment]
GUYS SAME PROBLEM, WHEN I OPEN THE PUSH NOTIFICATION MSG ON ANDROID DEVICE, 

I GET THIS ERROR 
[OneSignal] Error Getting Google Registration ID
[OneSignal] java.io.IOException: INVALID_PARAMETERS
[OneSignal] 	at com.google.android.gms.gcm.GoogleCloudMessaging.zza(Unknown Source)
[OneSignal] 	at com.google.android.gms.gcm.GoogleCloudMessaging.register(Unknown Source)
[OneSignal] 	at com.onesignal.PushRegistratorGPS$2.run(PushRegistratorGPS.java:151)
[OneSignal] 	at java.lang.Thread.run(Thread.java:761)

CODE: 
OneSignal.Current.SetLogLevel(LOG_LEVEL.VERBOSE, LOG_LEVEL.WARN);
OneSignal.Current.StartInit(Constants.OneSignalkey).Settings(new Dictionary<string, bool>() {
			{ IOSSettings.kOSSettingsKeyAutoPrompt, false },
			{ IOSSettings.kOSSettingsKeyInAppLaunchURL, true } })
			  .InFocusDisplaying(OSInFocusDisplayOption.Notification)
			  .HandleNotificationOpened((result) =>
			   {
				   Debug.WriteLine("HandleNotificationOpened: {0}", result.notification.payload.body);
			   })
			  .HandleNotificationReceived((notification) =>
			  {
				  Debug.WriteLine("HandleNotificationReceived: {0}", notification.payload.body);
			  })
			  .EndInit();

			OneSignal.Current.IdsAvailable((playerID, pushToken) =>
			{
				Debug.WriteLine("OneSignal.Current.IdsAvailable:D playerID: {0}, pushToken: {1}", playerID, pushToken);
			});


AS YOU SEE NOT DOING ANYTHING FANCY WHAT SEEMS TO BE THE PROBLEM?

**Comment by jkasten2** [comment]
@rabih-harb The logcat entry you passed above looks to the part of the same issue. This means our SDK couldn't ensure the Google registration id didn't change on app resume but it will still be able to check correctly on an app cold start.

The issue will be fixed in the next SDK update.

**Comment by rabih-harb** [comment]
Hello josh thank you for the heads up, 
I did find a work around for this, the issue is :  
This error appear when open the app then click on the android back button the app wasn't closed but minimized, when back button is closed onsignal says 
That curactivity variable is null , so that when u select the app again and the activity gets visible again u get this error. 
So as a solution am terminating the app when back button is clicked that way when u re open the app 
Everything is normal 
It's a bad hack but it is a temporary solution.

**Comment by corgalore** [comment]
I'm seeing this problem also; What's the ETA on the revised build?

**Comment by jkasten2** [comment]
The `java.io.IOException: INVALID_PARAMETERS` error will only affect the session count.

Is the `HandleNotificationOpened` callback having any issues firing when tapping on a notification?

**Comment by corgalore** [comment]
The callback does fire appropriately. That explains why my session stats
were off.

On Aug 9, 2017 7:01 PM, "Josh Kasten" <notifications@github.com> wrote:

The java.io.IOException: INVALID_PARAMETERS error will only affect the
session count.

Is the HandleNotificationOpened callback having any issues firing when
tapping on a notification?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/OneSignal/OneSignal-Xamarin-SDK/issues/35#issuecomment-321413633>,
or mute the thread

**Comment by lafritay** [comment]
@jkasten2 This causes a modal window to open and this error to be shown on our application.  Is there any way to work around this?  It's completely blocking shipping push notifications for us.

**Comment by jkasten2** [comment]
@lafritay Errors from the SDK will never be displayed to the screen unless you are calling [`setLogLevel`](https://documentation.onesignal.com/docs/xamarin-sdk#section--setloglevel-). Make sure this isn't called in your production app.

**Comment by lafritay** [comment]
@jkasten2 Ahh, that did the trick.  Thanks for the quick turnaround.

**Comment by jkasten2** [comment]
This is now fixed in the latest [OneSignal Xamarin 3.1.0 release](https://github.com/OneSignal/OneSignal-Xamarin-SDK/releases/tag/3.1.0)!

**Comment by ismaelgasparin** [comment]
tks
